src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    dateMap['YYYY'] =
+      dateMap['YY'] < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
```

Comment:
The conversion from 'YY' to 'YYYY' assumes that 'YY' will always be a number and does not handle cases where 'YY' could be malformed or non-numeric. It would be safer to parse the 'YY' part as an integer and then perform the comparison and concatenation.